core/variant/array.cpp

```diff
@@ -353,6 +353,36 @@ int Array::find(const Variant &p_value, int p_from) const {
 	return ret;
 }
 
+int Array::first_custom(const Callable& p_callable, int p_from = 0) const {
```

Comment:
This name feels misleading, using `first` makes it sound like it returns that value, not the index